pkg/apis/serving/fieldmask.go

```diff
@@ -183,6 +183,10 @@ func PodSpecMask(ctx context.Context, in *corev1.PodSpec) *corev1.PodSpec {
 	out.Volumes = in.Volumes
 	out.ImagePullSecrets = in.ImagePullSecrets
 	out.EnableServiceLinks = in.EnableServiceLinks
+	// Only allow setting AutomountServiceAccountToken to false
+	if !*in.AutomountServiceAccountToken && in.AutomountServiceAccountToken != nil {
```

Comment:
Now it's backwards, will still panic. The `&&` short-circuits, so if you turn this around, the second `!*in.AutomountServiceAccountToken` won't even be executed if `in.AutomountServiceAccountToken != nil` is false.